03-Design-Patterns-OOP/Movies.js

```diff
@@ -0,0 +1,210 @@
+};
+
+Movie.prototype.setMinutes = function (minutes) {
+    this.minutes = minutes;
+};
+
+Movie.prototype.setRating = function (rating) {
+    this.rating = rating;
+};
+
+Movie.prototype.getTitle = function () {
+    return this.title;
+};
+
+Movie.prototype.getGenre = function () {
+    return this.genre;
+};
+
+Movie.prototype.getMinutes = function () {
+    return this.minutes;
+};
+
+Movie.prototype.getRating = function () {
+    return this.rating;
+};
+
+Movie.prototype.play = function () {
+    return "Playing " + this.title;
+};
+
+Movie.prototype.stop = function () {
+    return "Stop playing " + this.title;
+};
+
+Movie.prototype.setActor = function (aux) {
+    this.actor.push(aux);
+    console.log("Se ha agregado el actor " + aux.get("name") + " correctamente");
+};
+
+Movie.prototype.getActor = function () {
+    var cadena = "Los actores de la pelicula son: ";
+    for (var i = 0; i < this.actor.length; i++) {
+        cadena += this.actor[i].get("name");
+        if (i < this.actor.length - 1) {
+            cadena += ", ";
+        }
+        else {
+            cadena += ".";
+        }
+    };
+    return console.log(cadena);
+};
+
+function MovieObserver() {
+
+};
+
+//Module Pattern
+function MovieModule() {
+    var title;
+    var minutes;
+    var genre;
+    var rating;
+
+    return{
+        setTitle: function (aux) {
+            title = aux;
+        },
+        getTitle: function () {
+            return title;
+        },
+        setMinutes: function (aux) {
+            minutes = aux;
+        },
+        getMinutes: function () {
+            return minutes;
+        },
+        setGenre: function (aux) {
+            genre = aux;
+        },
+        getGenre: function () {
+            return genre;
+        },
+        setRating: function (aux) {
+            rating = aux;
+        },
+        getRating: function () {
+            return rating;
+        }
+    };
+};
+
+function DownloadableMovie(title, genre, minutes, rating) {
+    Movie.call(this, title, genre, minutes, rating);
+};
+
+DownloadableMovie.prototype.download = function () {
+    console.log("Download method");
+};
+
+//Funcion que encontre en http://javascriptissexy.com/oop-in-javascript-what-you-need-to-know/
+//para utilizar en vez de extend()
+function inheritPrototype(childObject, parentObject) {
+    var copyOfParent = Object.create(parentObject.prototype);
+
+    copyOfParent.constructor = childObject;
+
+    childObject.prototype = copyOfParent;
+};
+
+var Social = function () {
+};
+
+Social.prototype = {
+    share: function (friendName) {
+        console.log("Sharing " + this.title + " with " + friendName);
+    },
+    like: function () {
+        console.log("Like");
+    }
+};
+
+//No pude aplicar extend() asi que utilize un funcion augment() que encontre en
+//la seccion The Mixin Pattern en http://addyosmani.com/resources/essentialjsdesignpatterns/book/
+function augment(receivingClass, givingClass) {
```

Comment:
This is good for function constructors, but not really for objects which are not meant to be instantiated.
Here is one other possible solution: https://gist.github.com/nicolasronsmansglobant/39ed74b7ad5355711931